theano/sandbox/cuda/basic_ops.py

```diff
@@ -3449,7 +3449,14 @@ def c_code(self, node, name, inp, out, sub):
         fail = sub['fail']
         str = """
         {
-            if (CudaNdarray_is_c_contiguous(%(input)s)){
+            bool strides_all_positive = False;
+            for (int i = 0; i < CudaNdarray_NDIM(%(input)s); i++){
+                if (CudaNdarray_HOST_STRIDES(%(input)s)[i] < 0){
+                    strides_all_positive = True;
+                    break;
+                }
+            }
+            if (CudaNdarray_is_c_contiguous(%(input)s) && strides_all_positive){
```

Comment:
You should't keep the `CudaNdarray_is_c_contiguous()` call in the condition.  We are replacing that.